rfcs/text/enzyme-to-rtl.md

```diff
@@ -0,0 +1,162 @@
+
+An [issue was raised](https://github.com/enzymejs/enzyme/issues/2429) in August 2020 to add support for React 17 to Enzyme, but as of January 2022 this has still not been added and it doesn't look like it will be for the foreseeable future.
+
+This is preventing us from upgrading Carbon to use React 17 and supporting it for Carbon's consumers. React 17 was released in October 2020, and [React 18](https://reactjs.org/blog/2021/06/08/the-plan-for-react-18.html) is now in beta so we are at risk of falling behind, which is not acceptable for a component library supporting so many other projects. We have already had a number of issues raised about the lack of React 17 support in our components.
+
+The logical solution for us is to move towards the officially recommended tesing utility for React, [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/). This has much better support, and is actually already used by most of the modern Sage projects we support.
+
+# Basic example
+
+The React Testing Library docs have a great [migration from Enzyme](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme) guide. This has a section on [basic Enzyme to React Testing Library migration examples](https://testing-library.com/docs/react-testing-library/migrate-from-enzyme#basic-enzyme-to-react-testing-library-migration-examples) which show exactly how the two libraries compare. I will add one of the basic examples here, but for a more in depth look, read the above docs.
+
+A simple Enzyme test for one of our components might look something like this:
+
+```jsx
+describe("Example Component", () => {
+  it("renders the title in a h1 element", () => {
+    const wrapper = shallow(<ExampleComponent title="Enzyme test" />);
+    expect(wrapper.find("h1").text()).toEqual("Enzyme test");
+  });
+});
+```
+
+The same test converted to React Testing Library will look like this:
+
+```jsx
+describe("Example Component", () => {
+  test("renders the title in a h1 element", () => {
+    render(<ExampleComponent title="Enzyme test" />);
+    expect(screen.getByRole("heading", { level: 1 })).toHaveTextContent(
+      "Enzyme test"
+    );
+  });
+});
+```
+
+The two libraries have slightly different APIs, but the tests are fairly similar and the effort required to convert them should be minimal.
+
+# Motivation
+
+The main motivation for us making this conversion is that, as previously mentioned, Enzyme does not yet support React 17. It is well over a year after it was released, and it does not look like support will be added in the near future. React 18 is in beta and we we be blocked from upgrading to that as well when it is released. It could be years before Enzyme add support for that, if ever.
```

Comment:
Looks great @samtjo! 🚀  This looks very comprehensive and the migration docs look useful. Just found this one spelling mistake - sorry to be that guy! Should be `we will be blocked`.